### PR TITLE
WIP: Parser update & command signing

### PR DESCRIPTION
**Pull request by DrJosh9000**
There are like 5 different ideas here; I intend to break this into multiple smaller PRs.

### CODE REVIEW

File: internal/pipeline/step.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+	}
+	if err := n.Decode(&full); err != nil {
+		return err
+	}
+
+	// Normalise into Command.
+	c.Command = full.Command
+	if c.Command == "" {
+		c.Command = strings.Join(full.Commands, "\n")
+	}
```
Comment (moskyb) [inline internal/pipeline/step.go:83]: i wonder if we should think about doing this the other way around - ie have `commands` be the "canonical" representation of commands. 

my reasoning here is that the `command` -> `commands` normalisation eg:
```yaml
command: >
  echo hello
  echo there
```
to 
```yaml
commands: ["echo hello", "echo there"]
```
feels like it retains more specificity than the other way around.

Comment (DrJosh9000) [reply internal/pipeline/step.go:83]: Thinking through this...

We upload `command` or `commands` but get back `BUILDKITE_COMMAND`. 

It should be easy to ensure consistency when computing signatures for `command` (1 string) and `BUILDKITE_COMMAND` (1 string). 

If we split `command` into `commands` (N strings), then we need to split `BUILDKITE_COMMAND` into the same N strings for the signatures to be consistent. If the user supplies `commands` and an element of `commands` contains multiple lines then this won't be consistent - we'd need to normalise by splitting each element of `commands`. Or we'd need to `strings.Join(commands, "\n")` solely for the benefit of computing signatures.

Comment (moskyb) [reply internal/pipeline/step.go:83]: yep, that makes total sense to me 👍 very happy to leave as is

**Comment by DrJosh9000** [comment]
Superseded by #2200 (and the work that led up to it).

**Pull request closed without merging**